src/components/molecules/Forms/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,127 @@
+import * as classNames from 'classnames'
+import * as React from 'react'
+import { Link } from 'react-router'
+
+import { Input } from 'components/atoms/Input/Input'
+import { InputPassword } from 'components/atoms/InputPassword/InputPassword'
+import { Form } from 'components/molecules/Form/Form'
+
+import './SignUp.scss'
+
+interface SignUpProps {
+  readonly onSubmit: (event: any) => any
+  readonly disabledButton?: boolean
+  readonly serverErrors?: any
+  readonly className?: string
+}
+
+export class SignUp extends React.Component<SignUpProps, undefined> {
+  private mutableEmailInput: HTMLInputElement
+  private mutablePasswordInput: HTMLInputElement
+  private mutableForm: HTMLFormElement
+
+  componentWillReceiveProps(newProps: any): void {
+    if (newProps.serverErrors.status) {
+      const { message } = newProps.serverErrors
+
+      if (message.includes('Password Requirements')) {
+        const { mutablePasswordInput } = this
+        mutablePasswordInput.setCustomValidity(newProps.serverErrors.message)
+        mutablePasswordInput.focus()
+      }
+
+      if (message.includes('The specified account already exists.')) {
+        const { mutableEmailInput } = this
+        mutableEmailInput.setCustomValidity(newProps.serverErrors.message)
+        mutableEmailInput.focus()
+      }
+
+      this.mutableForm.reportValidity()
+    }
+  }
+
+  readonly onValidate = (data: any, elements: any): boolean => {
+    const { password, confirmPassword } = data
+
+    if (password !== confirmPassword) {
+      elements.confirmPassword.setCustomValidity(`Passwords Don't Match`)
+      return false
+    }
+
+    return true
+  }
+
+  readonly onChangeRepeatPassword = (e: any, data: any, elements: any): void => {
+    const value = e.target.value
+    const { password, confirmPassword } = data
+
+    if (value !== '' && password !== confirmPassword)
+      elements.confirmPassword.setCustomValidity(`Passwords Don't Match`)
+
+    if (password === confirmPassword) elements.confirmPassword.setCustomValidity('')
+
+    if (value === '') elements.confirmPassword.setCustomValidity('')
+  }
+
+  readonly onChangeEmail = (e: any): void => {
```

Comment:
Just curious — can this be typed as a React Synthetic Event rather than any?

Reply:
It lets me if I cast the target type afterward:

```tsx
readonly onChangeEmail = (e: React.SyntheticEvent): void => {
    const input = e.target as HTMLInputElement;
    input.setCustomValidity('')
  }
```
and 
```tsx
readonly onChangeRepeatPassword = (e: React.SyntheticEvent, data: any, elements: any): void => {
    const target = e.target as HTMLInputElement;
    const value = target.value;
```

Reply:
That looks good!

Reply:
OK, I went and checked an old project of mine and then remembered the correct way to do this.

`SyntheticEvent` takes a generic parameter than can be used to type `currentTarget`. 

I think it would look like this:

```ts
readonly onChangeRepeatPassword = (e: React.SyntheticEvent<HTMLInputElement>, data: any, elements: any): void => {
    const target = e.currentTarget;
    const value = target.value;
```

There's also many more-specific interfaces that extend from SyntheticEvent, such as `KeyboardEvent` and `MouseEvent`.

Here's the [source code of the React typings](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L567).

Here's a [`target` vs `currentTarget`](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/12239) related discussion. 

And here's [documentation on `currentTarget`](https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget).

For our current use case I don't think we really need to change anything in this PR, but I wanted to leave this here as some sort of documentation on the topic.

I believe in onChange events using target and currentTarget has the same effect, and using the generic parameter of SyntheticEvent or casting doesn't make any real different either (since neither do any actual type checking).

@kennylavender